_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+In this step we will download our source code, perform all **test** which can be done on the source code level and produce **artifact** which 
+contains all elements required to start and run our application.
+
+This build configuration is not much different then usual step in a common Continuous Deployment pipeline.
+Alongside common parameters we have to define **artifacts** which will be generated after each execution. We are going to use them as base for next steps in the pipeline 
+
+In **TeamCity** we define **artifacts** by defining paths to files from working directory (which is created for each run of a configuration).
+Working directory is combination of files downloaded from a version control system and files generated during execution of build steps.
+We will define later elements of this directory. 
+
+We need two artifacts. The first one is `Dockerfile`. We already prepared this file in our source code and it will be stored in `docker` directory.
+The next file is ***tar*** file which will be generated by our **Gradle** build. It contains script which allows to execute our code and all required libraries. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_config.png" alt="Build Configuration" title="Build Configuration" class="img img-responsive style-screengrab">
+
+Now we are ready to instruct our build configuration how to download our source code. We use github as our code repository.
+We just have to choose type as *Git* and provide url to our application.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_vcs.png" alt="Version Control Configuration" title="Version Control Configuration" class="img img-responsive style-screengrab">
+
+We have the source code, so we can describe what to do with this code. As I mention before we will **test** our application and generate an **artifact** with executable content.   
+
+**TeamCity** have predefined runners for different build tools. In this case we are going to use **Gradle** runner to **test** and **build**. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_step.png" alt="Gradle Step" title="Gradle Step" class="img img-responsive style-screengrab">
+
+Now we can run our build. As result of we should see our **artifacts** in *"Artifact"* tab. 
+ 
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_result.png" alt="Build Result" title="Build Result" class="img img-responsive style-screengrab">
+
+## Release Image
+
+Our code is no longer needed. We have all we need to build the docker image. Now we have create our image and **release** with the right version. To simplify our example we are going to use current build number to define an image version.
+Next we will generate file with this version. This file give us possibility to pass information about version to the next steps.  
```

Comment:
generate a file with this version. This file will give us the possibility to pass information about the version